### PR TITLE
Run formatter on entire repo

### PR DESCRIPTION
**Pull request by collindutter**
We recently integrated the [`black`](https://black.readthedocs.io/en/stable/index.html) code formatter into the repo. I ran `poetry run black griptape` to bring the entire repo in-line with it.

We should use this PR to discuss whether we like these formatting changes, or should tweak the configuration settings. Though it should be noted that `black` is fairly opinionated about its' [style guidelines](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-black-code-style).

### CODE REVIEW

**Comment by andrewfrench** [comment]
All for consistency in code styling! It looks like we might need to run this again on the the tip of `main` to avoid a lot of conflicts.

**Comment by collindutter** [comment]
Haha yeah I suspected this PR was going to be a merge conflict nightmare. Since most commits to main will cause a merge conflict, let's all agree on the format and then I can fix all at once.

**Comment by andrewfrench** [comment]
I'm happy to give up some agency for better consistency in formatting (unsurprising coming from Go, which ships with a fairly opinionated out-of-the-box formatting tool). I'm not familiar with all of Python formatting options out there, but after a quick look around it seems like Black is pretty popular and the style choices seem reasonable to me.

I think this is a good baseline to define now and if we run into a problem we can change directions later without too much trouble.

**Review by andrewfrench — APPROVED** [review]

**Review by andrewfrench — APPROVED** [review]

**Comment by andrewfrench** [comment]
New Github action workflow! 🥳

**Review by andrewfrench — DISMISSED** [review]

File: griptape/artifacts/__init__.py [hunk 1]
```diff
@@ -14,5 +14,5 @@
     "TextArtifact",
     "BlobArtifact",
     "CsvRowArtifact",
-    "ListArtifact"
+    "ListArtifact",
```
Comment (vasinov) [inline griptape/artifacts/__init__.py:17]: Can we disable this rule? I think this can be sketchy considering how Python interprets tuples

**Review by vasinov — CHANGES_REQUESTED** [review]

File: griptape/chunkers/base_chunker.py [hunk 2]
```diff
@@ -65,14 +73,30 @@ def _chunk_recursively(self, chunk: str, current_separator: Optional[ChunkSepara
...
+                            separator.value
+                            + separator.value.join(
+                                subchanks[balance_index + 1 :]
+                            )
+                        )
                     else:
-                        first_subchunk = separator.value.join(subchanks[:balance_index + 1]) + separator.value
-                        second_subchunk = separator.value.join(subchanks[balance_index + 1:])
+                        first_subchunk = (
+                            separator.value.join(subchanks[: balance_index + 1])
```
[the remark this answers is not in the document]
Comment (collindutter) [reply griptape/chunkers/base_chunker.py:87]: Here is the [justification](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#slices). I'll see if we can disable.

File: griptape/artifacts/__init__.py (hunk 1 above)
Comment (collindutter) [reply griptape/artifacts/__init__.py:17]: Disabled

File: griptape/chunkers/base_chunker.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (collindutter) [reply griptape/chunkers/base_chunker.py:87]: It looks like black [does not support](https://github.com/psf/black/issues/367) disabling this as it is recommended by PEP 8. If this is a deal breaker, perhaps we need to look towards other formatters that have [more knobs](https://github.com/google/yapf#spaces_around_subscript_colon).

[the remark this answers is not in the document]
Comment (vasinov) [reply griptape/chunkers/base_chunker.py:87]: > It recommends no spaces around : operators for “simple expressions” (ham[lower:upper]), and extra space for “complex expressions” (ham[lower : upper + offset]).

Interesting...didn't know that. Let's follow this recommendation, even though it's ambiguous, IMO.

**Review by vasinov — APPROVED** [review]
Let's resolve conflicts and merge.

File: griptape/chunkers/base_chunker.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (collindutter) [reply griptape/chunkers/base_chunker.py:87]: Another potential code smell highlighter. Maybe we shouldn't be doing extra operations within a slice.

**Review by vasinov — APPROVED** [review]

**Pull request merged**